src/brogue/Architect.c

```diff
@@ -3540,8 +3540,19 @@ void restoreItem(item *theItem) {
 
     if (theItem->flags & ITEM_PREPLACED) {
         theItem->flags &= ~ITEM_PREPLACED;
-        getQualifyingLocNear(loc, *x, *y, true, 0, (T_OBSTRUCTS_ITEMS | T_AUTO_DESCENT | T_IS_DEEP_WATER | T_LAVA_INSTA_DEATH),
-                             (HAS_MONSTER | HAS_ITEM | HAS_STAIRS), true, false);
+
+        if (theItem->category == AMULET || theItem->category == KEY) {
+            // keep keys and the amulet as safe as before (but why would anyone throw them down a chasm?!)
```

Comment:
Tossing the amulet on a mastery run makes sense to me, the warden ignores it.

Reply:
The amulet should also simply float on lava, if we wanted to consider dropping this corner case.

Reply:
Agreed, the amulet is indestructible and I don't think keys need any special protection -- a key burning up in lava or falling through successive chasms seem like fair consequences for treating it carelessly.

Reply:
Thank you both for the feedback. I removed the special case, after verifying that the amulet is indeed immune to lava.

Reply:
:+1: may want to update changes file following this

Reply:
Missed that, thanks for catching it.